x-pack/agent/pkg/fleetapi/actions.go

```diff
@@ -0,0 +1,164 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package fleetapi
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// Action base interface for all the implemented action from the fleet API.
+type Action interface {
+	fmt.Stringer
+	Type() string
+	ID() string
+}
+
+// BaseAction is the base of all actions to be executed.
+type BaseAction struct {
+	ActionID   string
+	ActionType string
+}
+
+// Type returns the action type.
+func (a *BaseAction) Type() string {
+	return a.ActionType
+}
+
+// ID returns the action ID.
+func (a *BaseAction) ID() string {
+	return a.ActionID
+}
+
+// UnknownAction is an action that is not know by the current version of the Agent and we don't want
+// to return an error at parsing time but at execution time we can report or ignore.
+//
+// NOTE: We only keep the original type and the action id, the payload of the event is dropped, we
+// do this to make sure we do not leak any unwanted information.
+type UnknownAction struct {
+	*BaseAction
+	originalType string
+}
+
+// Type returns the type of the Action.
+func (a *UnknownAction) Type() string {
+	return "UNKNOWN"
```

Comment:
do you think it is worth to const it and export?

Reply:
Not really for a single type and its mostly for log reporting / debugging.